generators/newwebapp/index.js

```diff
@@ -185,8 +186,51 @@ module.exports = class extends Generator {
             };
 
             try {
-                const _fs = await generateFreestyleTemplate(this.destinationPath(sModuleName), FreestyleApp);
-                await _fs.commit();
+                if (this.options.enableFPM) {
+                    await ui5Writer.generate(this.destinationPath(sModuleName), {
+                        app: FreestyleApp.app,
+                        package: FreestyleApp.package,
+                        appOptions: FreestyleApp.appOptions
+                    }, this.fs);
+                } else {
+                    await generateFreestyleTemplate(this.destinationPath(sModuleName), FreestyleApp, this.fs);
+                    // make @sap-ux/fiori-freestyle-writer's MainView.controller
+                    // aware of easy-ui5's base controller
+                    const MainViewController = {
+                        js: this.destinationPath(sModuleName, "webapp/controller/MainView.controller.js")
+                    };
+                    await this.fs.write(
+                        MainViewController.js,
+                        (await this.fs.read(MainViewController.js))
+                            .toString()
+                            .replace(/sap\/ui\/core\/mvc\/Controller/g, "./BaseController")
+                    );
+                    this.log(
+                        `  ${chalk.blueBright(
+                            "\u26A0 \uFE0F patched @sap-ux's"
+                        )} MainViewController.js to use ./BaseController`
+                    );
+                    // fix up @sap-ux/fiori-freestyle-writer's test/flpSandbox.html -
+                    // sap.ushell is only available in sapui5
+                    // bootstrap only from there, no matter the used framework choice..
+                    const flpSandbox = { html: this.destinationPath(sModuleName, "webapp/test/flpSandbox.html") };
+                    await this.fs.write(
+                        flpSandbox.html,
+                        (await this.fs.read(flpSandbox.html))
+                            .toString()
+                            .replace(/src="(..)\/(test-)?resources/g, (match) => {
+                                return match.replace("..", "https://sapui5.hana.ondemand.com");
```

Comment:
BTW: if there is any URL to `sapui5.hana.ondemand.com` we should also replace it with `ui5.sap.com`.

Reply:
Agree but that shouldn't be in this PR.